lib/tools.js

```diff
@@ -2186,11 +2186,30 @@ function getInstances(adapter, objects, withObjects, callback) {
     });
 }
 
+/**
+ * Executes a command asynchronously. On success, the promise resolves with stdout and stderr.
+ * On error, the promise rejects with the exit code or signal, as well as stdout and stderr.
+ * @param {string} command The command to execute
+ * @param {import("child_process").ExecOptions} [execOptions] The options for child_process.exec
+ * @returns {import("child_process").ChildProcess & Promise<{stdout?: string; stderr?: string}>}
+ */
+function execAsync(command, execOptions) {
+    const defaultOptions = {
+        // we do not want to show the node.js window on Windows
+        windowsHide: true,
+        // And we want to capture stdout/stderr
+        encoding: 'utf8'
+    };
+    // @ts-ignore We set the encoding, so stdout/stdrr must be a string
+    return require('promisify-child-process').exec(command, { ...defaultOptions, ...execOptions });
```

Comment:
Really always a new require?

Reply:
require is cached, so it doesn't matter. But I moved it up.